src/core/corefile.cpp

```diff
@@ -84,30 +84,31 @@ void CoreFile::sendAvatarFile(uint32_t friendId, const QByteArray& data)
     QMutexLocker<QRecursiveMutex> locker{coreLoopLock};
 
     uint64_t filesize = 0;
-    uint8_t* file_id = nullptr;
-    uint8_t* file_name = nullptr;
+    std::vector<uint8_t>* file_id = nullptr;
+    std::vector<uint8_t>* file_name = nullptr;
     size_t nameLength = 0;
-    uint8_t avatarHash[TOX_HASH_LENGTH];
+    std::vector<uint8_t> avatarHash(tox_hash_length());
     if (!data.isEmpty()) {
-        static_assert(TOX_HASH_LENGTH <= TOX_FILE_ID_LENGTH,
-                      "TOX_HASH_LENGTH > TOX_FILE_ID_LENGTH!");
-        tox_hash(avatarHash, reinterpret_cast<const uint8_t*>(data.data()), data.size());
+        Q_ASSERT(tox_hash_length() <= tox_file_id_length());
+        tox_hash(avatarHash.data(), reinterpret_cast<const uint8_t*>(data.data()), data.size());
         filesize = data.size();
-        file_id = avatarHash;
-        file_name = avatarHash;
-        nameLength = TOX_HASH_LENGTH;
+        file_id = &avatarHash;
+        file_name = &avatarHash;
+        nameLength = tox_hash_length();
     }
     Tox_Err_File_Send error;
-    const uint32_t fileNum = tox_file_send(tox, friendId, TOX_FILE_KIND_AVATAR, filesize, file_id,
-                                           file_name, nameLength, &error);
+    const uint32_t fileNum =
+        tox_file_send(tox, friendId, TOX_FILE_KIND_AVATAR, filesize,
+                      file_id == nullptr ? nullptr : file_id->data(),
+                      file_name == nullptr ? nullptr : file_name->data(), nameLength, &error);
```

Comment:
I don't like pointer to std::vector. I don't think we even need them.

Also I think we should simplify this in the future. There is no good reason to have a "complex" file name.

Reply:
How about this?